app/src/shared/hooks/useWeb3Status.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+import { useState, useEffect, useCallback, useMemo } from 'react'
+
+import type { ErrorInUi } from '$shared/flowtype/common-types'
+import getWeb3, { validateWeb3 } from '$shared/web3/web3Provider'
+import { WalletLockedError } from '$shared/errors/Web3'
+import type { Address } from '$shared/flowtype/web3-types'
+
+import Web3Poller from '$shared/web3/web3Poller'
+
+import useIsMounted from './useIsMounted'
+
+type Result = {
+    requireWeb3: boolean,
+    web3Error: ?ErrorInUi | ?Error,
+    checkingWeb3: boolean,
+    isLocked: boolean,
+    account: ?Address,
+}
+
+export function useWeb3Status(requireWeb3: boolean = true): Result {
+    const [web3Error, setWeb3Error] = useState(null)
+    const [checkingWeb3, setCheckingWeb3] = useState(false)
+    const isMounted = useIsMounted()
+    const [account, setAccount] = useState(null)
+
+    useEffect(() => {
+        if (!requireWeb3 || checkingWeb3) { return () => {} }
```

Comment:
AFAIK no need to return empty function here

Reply:
oh is this because of flow? ugh.

Reply:
Yep 🤢